__tests__/pages/signup.js

```diff
@@ -1,29 +1,198 @@
+    const { getByTestId } = render(<Signup />)
     getByTestId('email')
     getByTestId('username')
     getByTestId('password')
     getByTestId('firstName')
     getByTestId('lastName')
   })
 
-  test('Should not submit values', async () => {
-    const { getByTestId } = render(<Signup {...props} />)
+  //to be readded once validation of form takes place in real time
+  // test('Should not submit values', async () => {
+  //   const { getByTestId } = render(<Signup {...props} />)
+  //   const submitButton = getByTestId('submit')
+  //   fireEvent.click(submitButton)
+  //   await wait(() => expect(submitSignup).not.toBeCalled())
+  // })
+
+  test('Should submit signup form values and redirect upon successful submission', async () => {
+    signupHelper.signupUser = jest
+      .fn()
+      .mockReturnValue(Promise.resolve({ success: true }))
+    Router.push = jest.fn()
+    const { getByTestId } = render(<Signup />)
+    const emailField = getByTestId('email')
+    const usernameField = getByTestId('username')
+    const passwordField = getByTestId('password')
+    const firstNameField = getByTestId('firstName')
+    const lastNameField = getByTestId('lastName')
     const submitButton = getByTestId('submit')
-    fireEvent.click(submitButton)
-    await wait(() => expect(submitSignup).not.toBeCalled())
+
+    await wait(
+      () =>
+        fireEvent.change(emailField, {
+          target: {
+            value: 'email@domain.com'
+          }
+        }),
+      fireEvent.change(usernameField, {
+        target: {
+          value: 'user name'
+        }
+      }),
+      fireEvent.change(passwordField, {
+        target: {
+          value: 'password123'
+        }
+      }),
+      fireEvent.change(firstNameField, {
+        target: {
+          value: 'user'
+        }
+      }),
+      fireEvent.change(lastNameField, {
+        target: {
+          value: 'name'
+        }
+      })
+    )
+
+    await wait(() => {
+      fireEvent.click(submitButton),
+        expect(signupHelper.signupUser).toHaveBeenCalledTimes(1),
+        expect(Router.push).toHaveBeenCalledWith('/signup/success')
+    })
   })
 
-  test('Should submit values', async () => {
-    const { getByTestId } = render(<Signup {...props} />)
+  test('Should submit signup form values and display error component upon failure of email', async () => {
+    signupHelper.signupUser = jest.fn().mockReturnValue(
+      Promise.resolve({
+        success: false,
+        errorMessage: {
+          confirmEmail: ['This email has been taken']
+        }
+      })
+    )
+    Router.push = jest.fn()
+    const { container, getByTestId } = render(<Signup />)
+    const emailField = getByTestId('email')
+    const usernameField = getByTestId('username')
+    const passwordField = getByTestId('password')
+    const firstNameField = getByTestId('firstName')
+    const lastNameField = getByTestId('lastName')
+    const submitButton = getByTestId('submit')
+
+    await wait(
```

Comment:
it would help alot for future developers if you created a `filloutSignupForm` function, so this logic is not duplicated (it is really long to be duplicated)